types/address.go

```diff
@@ -72,6 +74,31 @@ const (
 	Bech32PrefixConsPub = Bech32MainPrefix + PrefixValidator + PrefixConsensus + PrefixPublic
 )
 
+// cache variables
+var (
+	// AccAddress.String() is expensive and if unoptimized dominantly showed up in profiles,
+	// yet has no mechanisms to trivially cache the result given that AccAddress is a []byte type.
+	accAddrMu     sync.Mutex
+	accAddrCache  *simplelru.LRU
+	consAddrMu    sync.Mutex
+	consAddrCache *simplelru.LRU
+	valAddrMu     sync.Mutex
+	valAddrCache  *simplelru.LRU
+)
+
+func init() {
+	var err error
+	if accAddrCache, err = simplelru.NewLRU(25000, nil); err != nil {
```

Comment:
would be good to add some comments on how these numbers are chosen (copy paste what has been said in the comments here)

Reply:
sure, will add it. There was not much analysis behind it.